src/main/java/org/mapstruct/intellij/inspection/MapstructReferenceInspection.java

```diff
@@ -64,7 +68,14 @@ private boolean shouldRegisterProblem(BaseReference reference) {
             if ( reference instanceof BaseValueMappingReference valueMappingReference ) {
                 return valueMappingReference.getEnumClass() != null;
             }
-            return true;
+
+            return !containingClassIsAnnotationType( reference.getElement() );
+        }
+
+        private boolean containingClassIsAnnotationType(PsiElement element) {
+            return Optional.ofNullable( PsiTreeUtil.getParentOfType( element, PsiClass.class ) )
+                .map( PsiClass::isAnnotationType )
+                .orElse( false );
```

Comment:
There is really no need to use `ofNullable` here. I prefer assigning this to a value and doing a good old regular null check.